solr/core/src/java/org/apache/solr/api/V2HttpCall.java

```diff
@@ -434,7 +434,12 @@ protected void executeCoreRequest(SolrQueryResponse rsp) {
         // SolrCore counter
         core.close();
         core = null;
-        response.getHeaderNames().stream().forEach(name -> response.setHeader(name, null));
+        // Skip specific headers
+        // workaround for response.setHeader(name, null)
+        response.getHeaderNames().stream()
```

Comment:
This looks good for now IMO.

Longer term I'd love to see this code go away in favor of having the v2 API stuff get handled by its own servlet, but we're pretty far from that being workable for now.  It is a little hacky in the interim though 😦